### ISSUE TITLE
Performance issues in tpu/models/ (by P3)

### ISSUE THREAD

**Issue by DLPerf** [richardaecn/class-balanced-loss#22]
Hello! I've found a performance issue in tpu/models/: `batch()` should be called before `map()`, which could make your program more efficient. Here is [the tensorflow document](https://tensorflow.google.cn/guide/data_performance?hl=zh_cn#vectorized_mapping) to support it.

Detailed description is listed below:

- official/retinanet/dataloader.py/: `dataset.batch(batch_size, drop_remainder=True)`[(here)](https://github.com/richardaecn/class-balanced-loss/blob/1d7857208a2abc03d84e35a9d5383af8225d4b4d/tpu/models/official/retinanet/dataloader.py#L355) should be called before `dataset.map(_dataset_parser, num_parallel_calls=64)`[(here)](https://github.com/richardaecn/class-balanced-loss/blob/1d7857208a2abc03d84e35a9d5383af8225d4b4d/tpu/models/official/retinanet/dataloader.py#L353).
- official/retinanet/dataloader.py/: `dataset.batch(batch_size, drop_remainder=True)`[(here)](https://github.com/richardaecn/class-balanced-loss/blob/1d7857208a2abc03d84e35a9d5383af8225d4b4d/tpu/models/official/retinanet/dataloader.py#L450) should be called before `dataset.map(_dataset_parser, num_parallel_calls=64)`[(here)](https://github.com/richardaecn/class-balanced-loss/blob/1d7857208a2abc03d84e35a9d5383af8225d4b4d/tpu/models/official/retinanet/dataloader.py#L448).
- experimental/inception/inception_v3_old.py/: `.batch(batch_size)`[(here)](https://github.com/richardaecn/class-balanced-loss/blob/1d7857208a2abc03d84e35a9d5383af8225d4b4d/tpu/models/experimental/inception/inception_v3_old.py#L256) should be called before `.map(parser)`[(here)](https://github.com/richardaecn/class-balanced-loss/blob/1d7857208a2abc03d84e35a9d5383af8225d4b4d/tpu/models/experimental/inception/inception_v3_old.py#L256).

Besides, you need to check the function called in `map()`(e.g., `parser` called in `.map(parser)`) whether to be affected or not to make the changed code work properly. For example, if `parser` needs data with shape (x, y, z) as its input before fix, it would require data with shape (batch_size, x, y, z).

Looking forward to your reply. Btw, I am very glad to create a PR to fix it if you are too busy.

**Comment by DLPerf** [comment]
Hello, I'm looking forward to your reply~
